javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/JavaParserFacade.java

```diff
@@ -620,7 +620,7 @@ protected Node findContainingTypeDeclOrObjectCreationExpr(Node node) {
      * references an outer class -- as its ancestor, return the declaration corresponding to the class name specified.
      */
     protected Node findContainingTypeDeclOrObjectCreationExpr(Node node, String className) {
-        if (node instanceof ClassOrInterfaceDeclaration && ((ClassOrInterfaceDeclaration) node).getFullyQualifiedName().get().equals(className)) {
+        if (node instanceof ClassOrInterfaceDeclaration && ((ClassOrInterfaceDeclaration) node).getFullyQualifiedName().get().endsWith(className)) {
```

Comment:
I was thinking about this for #2928 too, which also uses `endsWith`... 

Presumably `SomeOtherObject` would match `endsWith("Object")`, meaning that something more sophisticated like splitting it then iterating right to left would be needed in order to do this robustly?

Reply:
I don't see a use case that wouldn't work with this algorithm but I could be wrong. I modified the unit tests by moving a class to test the reliability of the algorithm.

Reply:
I was thinking more along the lines of `endsWith` being too greedy, and accidentally matching something it shouldn't. 

e.g.
```
searching for endsWith("c.ClassName") matches:

a.b.c.ClassName
x.y.z.a.b.c.ClassName
a.b.abc.ClassName
```


```
searching for endsWith("List") matches:

java.util.List
java.util.ArrayList
com.github.javaparser.NodeList
com.meeeee.List
```

The "correct" reference for each will depend on the imports used, iirc.

I don't know enough about these tests to know if these are realistic concerns or not though, so please do feel free to tell me I'm wrong / don't need to worry about this! :+1:

Reply:
What you say seems to make sense, thanks for explaining! 

> If you think this is not a good approach I give up

As long as it works, I really don't care either way :wink: 

Worst case scenario is that either one gets merged and then it then gets improved upon later -- nice, chilled, and fairly low-stakes :)

Reply:
Ok i merge it and wait and see mister deadlocklogic finds a new use case that would not be well handled by this fix.